spark/src/main/scala/org/apache/comet/serde/CometMakeDecimal.scala

```diff
@@ -0,0 +1,41 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.comet.serde
+
+import org.apache.spark.sql.catalyst.expressions.{Attribute, MakeDecimal}
+import org.apache.spark.sql.types.DecimalType
+
+import org.apache.comet.serde.QueryPlanSerde.{exprToProtoInternal, optExprWithInfo, scalarFunctionExprToProtoWithReturnType}
+
+object CometMakeDecimal extends CometExpressionSerde[MakeDecimal] {
+  override def convert(
+      expr: MakeDecimal,
+      inputs: Seq[Attribute],
+      binding: Boolean): Option[ExprOuterClass.Expr] = {
+    val childExpr = exprToProtoInternal(expr.child, inputs, binding)
+    val optExpr = scalarFunctionExprToProtoWithReturnType(
+      "make_decimal",
+      DecimalType(expr.precision, expr.scale),
+      false,
```

Comment:
```suggestion
      !expr.nullOnOverflow,
```

Reply:
I see what changed now. Previously, we only supported `MakeDecimal` in the case that `nullOnOverflow=true`, but now we always support it. So this PR does introduce a functional change.